### DISCUSSION TITLE
Multiple vsg::view's and z-buffer

### DISCUSSION THREAD

**Discussion by dov** [vsg-dev/VulkanSceneGraph#1332] [General]
I'm having a trouble understanding the interaction between multiple views that you add to a render graph. My current render graph contains multiple views. The bottom view is my 3D scene, and on top of that I have various annotation views, e.g. a view for interactively drawing of a rectangle, a view containing an orientation gizmo, and right now I'm working on a view drawing rulers at the edges of the window.

Currently the views compete about the z-location. I.e. if the first view draws at z=1 and the second view draws at z=2, then the drawing from the second view will not appear on the screen.

My question is how I can make apply the "painters algorithm" from one view to the next so I won't have to manually assign increasing z-coordinates for each view? Or should I perhaps create a different renderGraph for each layer?

**Comment by robertosfield** [comment]
There are various approaches you can take for the overlaid views, the best approach depends upon what you are doing in each view.  You can disable depth test in the overlaid views, or clear the depth buffer before rendering.  Have a look at the vsgoverlay and vsgmultiviews examples.
